ansible/playbooks/deploy_moriod.yml

```diff
@@ -0,0 +1,118 @@
+---
+- name: Provision EC2 Instances and Configure DNS
+  hosts: localhost
+  gather_facts: no
+  collections:
+    - amazon.aws
+    - community.general
+  vars:
+    region: "eu-central-1"
+    subnet_id: "subnet-04faec9c5fe410b3e"
+    security_group_id: "sg-09a11726f166660ca"
+    zone_id: "Z04114953S6ELFC6IJG0T"  # Replace with your Route 53 hosted zone ID
+    ami_id: "ami-0552f4006d8da5f5f"  # Replace with your preferred AMI ID
+
+  tasks:
+    - name: Generate random DNS names for instances
+      set_fact:
+        dns_names: []
+        
+    - name: Generate random DNS names and append to list
+      set_fact:
+        dns_names: "{{ dns_names + [lookup('community.general.random_string', count=1, length=3, special=False, upper=False)] }}"
+      loop: "{{ range(node_count)|list }}"
+
+    - name: Create an SSH Key Pair
+      amazon.aws.ec2_key:
+        name: "morio_test_{{ pr_id }}-{{ item }}"
+        key_material: "{{ lookup('file', '~/.ssh/id_rsa.pub') }}"
+      loop: "{{ dns_names }}"
+
+    - name: Create EC2 instances
+      amazon.aws.ec2_instance:
+        name: "morio-test-{{ pr_id }}-{{ item }}"
+        key_name: "morio_test_{{ pr_id }}-{{ item }}"
+        instance_type: "t3.medium"
+        image_id: "{{ ami_id }}"
+        region: "{{ region }}"
+        vpc_subnet_id: "{{ subnet_id }}"
+        security_group: "{{ security_group_id }}"
+        network:
+          assign_public_ip: true
+        wait: yes
+        tags:
+          Name: "morio-test-{{ pr_id }}-{{ item }}"
+          morio: "true"
+          pull_request: "{{ pr_id }}"
+        user_data: |
+          #!/bin/bash
+          echo "ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIEe3xiWGZ6u7h4/Z+MHmvk81uYNVSTlXIGR1bpyUEoa6 admin@cert.europa.eu" >> /home/admin/.ssh/authorized_keys
```

Comment:
Let's put these SSH keys in secrets too ok

Reply:
ssh_pubkey_1: "{{ lookup('env', 'ITEST_AWS_SSH_PUBKEY_1_ID') }}"
    ssh_pubkey_2: "{{ lookup('env', 'ITEST_AWS_SSH_PUBKEY_2_ID') }}"
    
    I've used it from secrets too.